confgenerator/config/config.go

```diff
@@ -446,6 +456,9 @@ var (
 				return nil
 			},
 		},
+		"hostmetrics": collectionIntervalValidation,
```

Comment:
This is very overengineered.

Reply:
Agreed. Any concrete suggestions on how to improve it? FWIW, some of it may go away (or take another form) with the custom `yaml.Unmarshaler` implementation, so can we defer the cleanup to a later PR?

Reply:
Yup, we can defer this.